octopus-relay/src/appchain/state.rs

```diff
@@ -0,0 +1,397 @@
+use std::convert::TryInto;
+
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::{LazyOption, UnorderedMap, Vector};
+use near_sdk::{env, AccountId, Balance, BlockHeight, Timestamp};
+
+use crate::storage_key::StorageKey;
+use crate::types::{AppchainStatus, Delegator, LiteValidator, Validator, ValidatorSet};
+use crate::{AppchainId, DelegatorId, ValidatorId, VALIDATOR_SET_CYCLE};
+
+const INVALID_DELEGATORS_DATA_OF_VALIDATOR: &'static str = "Invalid delegators data of validator";
+
+/// Appchain delegator of an appchain validator
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct AppchainDelegator {
+    /// Id of appchain delegator
+    pub delegator_id: DelegatorId,
+    /// Account id of the delegator
+    pub account_id: AccountId,
+    /// Authorized balance of the delegator
+    pub amount: Balance,
+    /// Block height of the delegation of the delegator
+    pub block_height: BlockHeight,
+}
+
+/// Appchain validator of an appchain
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct AppchainValidator {
+    /// Id of appchain validator
+    pub validator_id: ValidatorId,
+    /// Account id of the validator
+    pub account_id: AccountId,
+    /// Staked balance of the validator
+    pub amount: Balance,
+    /// Block height which the validator started staking
+    pub block_height: BlockHeight,
+    /// Delegators of the validator
+    pub delegators: UnorderedMap<DelegatorId, LazyOption<AppchainDelegator>>,
+}
+
+/// Appchain state of an appchain of Octopus Network
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct AppchainState {
+    /// Id of the appchain
+    pub id: AppchainId,
+    /// Validators collection of the appchain
+    pub validators: UnorderedMap<ValidatorId, LazyOption<AppchainValidator>>,
```

Comment:
Got it.
Considering we can also remove validator from validator set, we can NOT ensure the order of validators in storage on chain even if we use an extra Vector to store their ids (because the mechanism of removing element from array is to move the last element to the position of removed one). 
So,  I'll add sorting logic when return validator set out of contract.